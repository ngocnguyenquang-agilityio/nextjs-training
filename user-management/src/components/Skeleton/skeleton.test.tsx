import { render } from '@testing-library/react';

// Components
import { UserTableSkeleton, TechTableSkeleton } from '.';

describe('UserTable Skeleton', () => {
  test('Should render successfully', () => {
    const comp = render(<UserTableSkeleton />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<UserTableSkeleton />);
    expect(comp).toMatchSnapshot();
  });
});

describe('TechTable Skeleton', () => {
  test('Should render successfully', () => {
    const comp = render(<TechTableSkeleton />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<TechTableSkeleton />);
    expect(comp).toMatchSnapshot();
  });
});
