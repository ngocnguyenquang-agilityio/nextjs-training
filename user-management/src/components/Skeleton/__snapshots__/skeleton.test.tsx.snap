// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechTable Skeleton Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="w-full text-sm text-left text-gray-500"
      >
        <thead
          class="text-xs text-gray-700 uppercase bg-gray-50"
        >
          <tr>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="w-full text-sm text-left text-gray-500"
    >
      <thead
        class="text-xs text-gray-700 uppercase bg-gray-50"
      >
        <tr>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserTable Skeleton Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="w-full text-sm text-left text-gray-500"
      >
        <thead
          class="text-xs text-gray-700 uppercase bg-gray-100"
        >
          <tr>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
            <th
              class="px-6 py-3"
              scope="col"
            >
              <div
                class="h-6 w-32 rounded bg-gray-200"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-32 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-32 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-32 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-32 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <div
                  class="h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-24 rounded bg-gray-100"
                />
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-32 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
                <div
                  class="h-[38px] w-[38px] rounded bg-gray-100"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="w-full text-sm text-left text-gray-500"
    >
      <thead
        class="text-xs text-gray-700 uppercase bg-gray-100"
      >
        <tr>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
          <th
            class="px-6 py-3"
            scope="col"
          >
            <div
              class="h-6 w-32 rounded bg-gray-200"
            />
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-32 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-32 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-32 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-32 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <div
                class="h-8 w-8 rounded-full bg-gray-100"
              />
              <div
                class="h-6 w-24 rounded bg-gray-100"
              />
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-32 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="h-6 w-16 rounded bg-gray-100"
            />
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
              <div
                class="h-[38px] w-[38px] rounded bg-gray-100"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
